internal/controller/nonadminbackup_controller_test.go

```diff
@@ -149,107 +148,108 @@ func deleteTestNamespaces(ctx context.Context, nonAdminNamespaceName string, oad
-			oadpNamespaceName = nonAdminNamespaceName + "-oadp"
-		}
+		ctx                     = context.Background()
+		nonAdminObjectName      string
+		nonAdminObjectNamespace string
+		oadpNamespace           string
+		veleroBackupNameUUID    string
+		counter                 = 0
 	)
-
+	ginkgo.BeforeEach(func() {
+		counter++
+		nonAdminObjectName = fmt.Sprintf("nab-object-%v", counter)
+		nonAdminObjectNamespace = fmt.Sprintf("test-nab-reconcile-%v", counter)
+		oadpNamespace = nonAdminObjectNamespace + "-oadp"
+		veleroBackupNameUUID = function.GenerateNacObjectNameWithUUID(nonAdminObjectNamespace, nonAdminObjectName)
+		gomega.Expect(createTestNamespaces(ctx, nonAdminObjectNamespace, oadpNamespace)).To(gomega.Succeed())
+	})
 	ginkgo.AfterEach(func() {
 		nonAdminBackup := &nacv1alpha1.NonAdminBackup{}
 		if k8sClient.Get(
 			ctx,
 			types.NamespacedName{
-				Name:      testNonAdminBackupName,
-				Namespace: nonAdminNamespaceName,
+				Name:      nonAdminObjectName,
+				Namespace: nonAdminObjectNamespace,
 			},
 			nonAdminBackup,
 		) == nil {
 			gomega.Expect(k8sClient.Delete(ctx, nonAdminBackup)).To(gomega.Succeed())
 		}
-
-		gomega.Expect(deleteTestNamespaces(ctx, nonAdminNamespaceName, oadpNamespaceName)).To(gomega.Succeed())
+		gomega.Expect(deleteTestNamespaces(ctx, nonAdminObjectNamespace, oadpNamespace)).To(gomega.Succeed())
 	})
-
 	ginkgo.DescribeTable("Reconcile triggered by NonAdminBackup Delete event",
 		func(scenario nonAdminBackupSingleReconcileScenario) {
-			updateTestScenario()
-
-			gomega.Expect(createTestNamespaces(ctx, nonAdminNamespaceName, oadpNamespaceName)).To(gomega.Succeed())
-
 			result, err := (&NonAdminBackupReconciler{
 				Client: k8sClient,
 				Scheme: testEnv.Scheme,
 			}).Reconcile(
 				context.Background(),
 				reconcile.Request{NamespacedName: types.NamespacedName{
-					Namespace: nonAdminNamespaceName,
-					Name:      testNonAdminBackupName,
+					Name:      nonAdminObjectName,
+					Namespace: nonAdminObjectNamespace,
 				}},
 			)
-
 			gomega.Expect(result).To(gomega.Equal(scenario.result))
 			gomega.Expect(err).To(gomega.Not(gomega.HaveOccurred()))
 		},
 		ginkgo.Entry("Should exit", nonAdminBackupSingleReconcileScenario{
 			result: reconcile.Result{},
 		}),
 	)
-
 	ginkgo.DescribeTable("Reconcile triggered by NonAdminBackup Create/Update events and by Requeue",
 		func(scenario nonAdminBackupSingleReconcileScenario) {
-			updateTestScenario()
-
-			gomega.Expect(createTestNamespaces(ctx, nonAdminNamespaceName, oadpNamespaceName)).To(gomega.Succeed())
-
-			nonAdminBackup := buildTestNonAdminBackup(nonAdminNamespaceName, scenario.spec)
-			gomega.Expect(k8sClient.Create(ctx, nonAdminBackup)).To(gomega.Succeed())
+			nonAdminBackup := buildTestNonAdminBackup(nonAdminObjectNamespace, nonAdminObjectName, scenario.nonAdminBackupSpec)
+			gomega.Expect(k8sClient.Create(ctx, nonAdminBackup.DeepCopy())).To(gomega.Succeed())
+			nonAdminBackupAfterCreate := &nacv1alpha1.NonAdminBackup{}
```

Comment:
this get call is unnecessary

Reply:
I don't remember now, but I think it was required, otherwise we were working with modified object after creation which caused wrong data to be compared against.

This one creates the object:
```
gomega.Expect(k8sClient.Create(ctx, nonAdminBackup.DeepCopy())).To(gomega.Succeed())
```

--> reconcile works on the object

---> We need to get object which was after reconcile, to continue working on it.